### PR TITLE
add pull request template

### PR DESCRIPTION
**Pull request by graphaelli**
As discussed, feedback very welcome.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+* unit tests
+* integration tests
+* documentation
+
+Major changes require a number of considerations including impact on:
+
+* logging selector(s)
+* metrics
+* telemetry
+* ESS/ECE
```
Comment (simitt) [inline .github/pull_request_template.md]: Might not be clear to everyone what ESS/ECE is. Without linking to some docs (might not exist yet) this could be hard to follow.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<!-- Thanks for sending a pull request!
+
+If this is your first contribution, please review and sign our contributor agreement -
+https://www.elastic.co/contributor-agreement.
+
+See also https://github.com/elastic/apm-server/blob/master/CONTRIBUTING.md for more tips on contributing.
+-->
+
+<!--
+Replace this comment with a description of what's being changed by this PR.
```
Comment (simitt) [inline .github/pull_request_template.md]: I'd ask for a description (what) and a motivation (why). My experience is the more structured the templates are the more information one provides.

**Comment by jalvz** [comment]
Good idea, but I would remove that every time I create the PR and we don't have that many contributors, so I'd rather put that on `CONTRIBUTING.md` (Just my 2c)

**Comment by simitt** [comment]
@graphaelli are we proceeding with this?

**Comment by graphaelli** [comment]
Yes, just haven't gotten back to it yet, thanks for the bump.

**Comment by jalvz** [comment]
will this show up in the backport pr's?

**Comment by graphaelli** [comment]
The ones created by the backport tool set the description directly - this should only impact prs created via the ui.
